plugins/pulp_rpm/plugins/migrations/0031_yum_metadata_storage_path.py

```diff
@@ -0,0 +1,39 @@
+from gettext import gettext as _
+import logging
+import os
+import tempfile
+
+from pulp.server.db import model
+from pulp_rpm.plugins.db import models
+
+
+_log = logging.getLogger('pulp')
+
+
+def migrate(*args, **kwargs):
+    """
+    Re-calculate storage path for yum_repo_metadata_file unit
+    """
+    metadata_files = models.YumMetadataFile.objects.filter()
```

Comment:
As a rule, we avoid using model code in migrations, and to the greatest extent possible, avoid using any code that's outside the migration. This is because that code can change, dependencies change, etc. which can lead to a scenario where this migration would behave differently a year from now than it does today.

So it is better to use pymongo directly, and copy into here whatever code is necessary to calculate new values for any fields that have to change.